app/controllers/shipit/api/base_controller.rb

```diff
@@ -7,6 +7,7 @@ class BaseController < ActionController::Base
       include Paginable
 
       rescue_from ApiClient::InsufficientPermission, with: :insufficient_permission
+      rescue_from EnvironmentVariables::NotPermitted, with: :insufficient_permission
```

Comment:
Hum, IMO non whitelisted env var should return a `422`. It's basically a validation error.